lib/index.js

```diff
@@ -2,18 +2,21 @@
 
 const onlySpreadCSS = require('./rules/only-spread-css');
 const noUnusedStyles = require('./rules/no-unused-styles');
+const onlyExtendDefined = require('./rules/only-extend-defined');
 
 module.exports = {
   rules: {
     'only-spread-css': onlySpreadCSS,
     'no-unused-styles': noUnusedStyles,
+    'only-extend-defined': onlyExtendDefined,
   },
 
   configs: {
     recommended: {
       rules: {
         'react-with-styles/only-spread-css': 'error',
         'react-with-styles/no-unused-styles': 'error',
+        'react-with-styles/only-extend-defined': 'error',
```

Comment:
FWIW, adding this to the recommended list makes this a breaking change. Sometimes we add rules at first and then when we are ready to make a breaking change, we lump all of the modifications to the recommended config into one breaking change.

However, given how rarely we work on this particular plugin, it might make sense to do this now.

Reply:
might still be worth it to ship it as minor first (without this line) and then do a followup as a major that enables it.